src/main/java/org/folio/persist/HoldingsRepository.java

```diff
@@ -2,20 +2,23 @@
 
 import static org.folio.rest.impl.HoldingsStorageAPI.HOLDINGS_RECORD_TABLE;
 import static org.folio.rest.persist.PgUtil.postgresClient;
+import static org.folio.rest.tools.utils.TenantTool.tenantId;
 
+import java.util.List;
 import java.util.Map;
 
 import org.folio.rest.jaxrs.model.HoldingsRecord;
-import org.folio.rest.persist.PostgresClient;
 
 import io.vertx.core.Context;
+import io.vertx.core.Future;
 
 public class HoldingsRepository extends AbstractRepository<HoldingsRecord> {
   public HoldingsRepository(Context context, Map<String, String> okapiHeaders) {
-    this(postgresClient(context, okapiHeaders));
+    super(postgresClient(context, okapiHeaders), HOLDINGS_RECORD_TABLE, HoldingsRecord.class,
+      tenantId(okapiHeaders));
   }
 
-  public HoldingsRepository(PostgresClient postgresClient) {
-    super(postgresClient, HOLDINGS_RECORD_TABLE, HoldingsRecord.class);
+  public Future<List<HoldingsRecord>> getAll() {
+    return postgresClientFuturized.get(tableName, new HoldingsRecord());
```

Comment:
Can this actually return all holding records? How about performance for large tenant?

Reply:
That's a good question, actually this method is used only when a deleteAll endpoint is called for instances/hildings/items (in this case we need to issue domain events for all the records). 

The delete all endpoint is really dangerous itself, and I'm not sure whether it is used by anyone except unit tests, maybe @marcjohnson-kint has an answer. My personal preference would be to mark the endpoints deprecated and remove them  at all. But if it is not an option, should we set a limit (e.g. 1000 records for the method)?

Reply:
> The delete all endpoint is really dangerous itself

Why do you consider it dangerous?

> and I'm not sure whether it is used by anyone except unit tests, maybe @marcjohnson-kint has an answer. 

Alas, this came up [recently](https://folio-project.slack.com/archives/CFQU1MF61/p1610456907318300?thread_ts=1610455157.316900&cid=CFQU1MF61) and there are folks using them.

> My personal preference would be to mark the endpoints deprecated and remove them at all. But if it is not an option, should we set a limit (e.g. 1000 records for the method)?

Yeah, I would like to remove them. I regret defining them in the early days. 

However, I believe part of the scope of the forthcoming batch APIs is to include a delete by CQL endpoint, which would likely have similar challenges as it's possible to define CQL that matches all records.

Reply:
> Why do you consider it dangerous?
Because it is easy to drop the whole table using them... Someone just may forgot to specify an id and the delete all API will be executed.

Reply:
Hi @marcjohnson-kint, @hjiebsco, how do you think we should proceed with this? We can 1) Deprecate API; 2) Implement some sort of a batching removal. 3) an another idea?

Reply:
I was concerned about the `getAll` performance because it did not use streaming. I do not think that we have to get rid of `deleteAll` if it is needed. The risk can be controlled by permissions. As for passing all deleted records to Kafka/elastic search, we can use either streaming or maybe consider other approaches, for example, only pass necessary info (id) or a single delete all message to delete all related elastic search indexes.

Reply:
I afraid that streaming won't help us here as well. We have to fetch all records **before** executing deleteAll and hold them in memory until receiving response from the deleteAll method (just for a case when we published events that records are removed but then DB operation is failed). Another concern, the streaming operation should lock records, as far as I know, so delete have to wait the lock to be released, there might be some issues with this.  

Probably we can fetch a portion of records (e.g. 1000) and then remove them by ids and then send events, repeat until any record exist.

The separate event for delete all action sounds really interesting, what do you think @MikhailFokanov?

Reply:
> We can 1) Deprecate API;

I don't think deprecating the API will help as it will still be available.

> 2) Implement some sort of a batching removal. 

What do you mean by batching? Select a batch of records, delete them and then publish messages for those?

> 3) an another idea?

Publish a specific message for the deletion of all records instead of trying to publish a message for each record.

Reply:
> Publish a specific message for the deletion of all records instead of trying to publish a message for each record.

@marc-johnson, it will break the Domain Event pattern. If item is deleted, all consumers should know, that it is deleted. And also it would be redundant complexity of supporting additional logic in all modules, which consume to this topic to handle this edge case, instead of just sending the standard messages.

@marc-johnson, As I got from the conversation in slack, that you mentioned: this endpoint is used by "one or two data migration folks". So maybe it is not the place for spending efforts on complicated optimizations. Actually, I don't see any business reason, why the migration folks cannot wait a bit longer for this particular request to be executed.

Reply:
> it will break the Domain Event pattern. If item is deleted, all consumers should know, that it is deleted.

If a single item is deleted why should *all* consumers know about it? I thought the current design meant that only the consumer in the group that had the partition for the related instance received the message. Have I misunderstood?

Or by consumers are you referring to different modules rather than different consumers in a consumer group?

> @marc-johnson, As I got from the conversation in slack, that you mentioned: this endpoint is used by "one or two data migration folks". So maybe it is not the place for spending efforts on complicated optimizations. Actually, I don't see any business reason, why the migration folks cannot wait a bit longer for this particular request to be executed.

I'm not sure it's appropriate for me to comment on this. I think it is the Product Owners who decide whether there is a business reason for something not us. Whether they can tolerate a longer latency is a question to ask them.

Reply:
For record deletion, the current standard message is to publish the full old record content, do we have use case where the consumer actually needs the full content, or maybe the id would be enough?
As for the "delete all" message, is not it a regular message in a message driven system where the publisher emits, and consumer can decide what type of messages to consume?

Reply:
An Id should be enough, but for Items we have to also fetch related holdings-records to get instance id because it is partition key for the topic. 

But mod-search is not the only consumer, mod-remote-storage is an another consumer, probably they need locationId of the item as well, am I correct @khandramai?

Reply:
https://issues.folio.org/browse/MODINVSTOR-664